### PR TITLE
Rust upgrade 1.2.0 nightly

### PR DESCRIPTION
**Pull request by pataroose**
Some cleanups, build image on readme.

Had an issue with rust-lang/rust#22649 in `tests/basic.rs` which was solved by adding `[..]` at the end of the test strings as well as changing some of the types. These can be changed back when the bug is fixed, but it should only affect testing.

### CODE REVIEW

**Comment by pataroose** [comment]
It looks like it's upset that I'm using unstable features on the stable release channel. Is there any chance the Travis compiler could be updated?

**Comment by tomaka** [comment]
Thanks!

> It looks like it's upset that I'm using unstable features on the stable release channel. Is there any chance the Travis compiler could be updated?

Could you add this to .travis.yml?
```
rust: nightly
```

File: package/src/lib.rs [hunk 1]
```diff
@@ -22,9 +22,7 @@ impl Package {
     ///
     /// If multiple resources have the same path, the first one will be returned. You should try
     ///  to avoid this situation.
-    pub fn find<P>(&self, resource: P) -> Option<&'static [u8]> where P: AsPath {
-        let resource = resource.as_path();
-
+    pub fn find(&self, resource: &Path) -> Option<&'static [u8]> {
```
Comment (tomaka) [inline package/src/lib.rs]: The `AsPath` trait has been replaced by `AsRef<Path>`.
Could you put `pub fn find<P>(&self, resource: P) -> Option<&'static [u8]> where P: AsRef<Path> {` instead? This allows writing `find("hello")` for example.

**Comment by pataroose** [comment]
Ah, thanks for that.

**Pull request merged**